src/app.ts

```diff
@@ -176,16 +173,74 @@ export async function newApp(
   // Setup routes
   //
 
-  app.get("/login", spidAuth);
-
+  await registerLoginRoute(app);
   registerPublicRoutes(app);
   registerAuthenticationRoutes(app, authenticationBasePath);
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
+  const timer = idpMetadataUpdater(
+    app,
+    container.resolve<number>(IDP_REFRESH_INTERVAL_SECONDS) * 1000
+  );
+  app.on("server:stop", () => {
+    clearInterval(timer);
+  });
   return app;
 }
 
+async function registerLoginRoute(app: Express): Promise<void> {
+  // Add the strategy to authenticate the proxy to SPID.
+  passport.use(
+    "spid",
+    await container.resolve<Promise<passport.Strategy>>(SPID_STRATEGY)
+  );
+  const spidAuth = passport.authenticate("spid", { session: false });
+  app.get("/login", spidAuth);
+}
+
+async function clearAndReloadSpidStrategy(
+  app: Express,
+  onRefresh?: () => void
+): Promise<void> {
+  log.info("Started Spid strategy re-initialization ...");
+
+  container.cache.delete(SPID_STRATEGY);
+
+  passport.unuse("spid");
+
+  // Remove /login route from Express router stack
+  // tslint:disable: no-any
+  // tslint:disable-next-line: no-object-mutation
+  app._router.stack = app._router.stack.filter((_: any) => {
+    if (
+      _.route &&
+      _.route.path === "/login" &&
+      _.route.methods &&
+      _.route.methods.get
+    ) {
+      return false;
+    }
+    return true;
+  });
+
+  await registerLoginRoute(app);
+  log.info("Spid strategy re-initialization complete.");
+  if (onRefresh) {
+    onRefresh();
+  }
+}
+
+export function idpMetadataUpdater(
+  app: Express,
+  refreshTime: number,
+  onRefresh?: () => void
+): NodeJS.Timer {
+  return setInterval(async () => {
```

Comment:
`setInterval` will not handle the failure of this promise, also an `async` function that just `awaits` a function that returns a promise is exactly the same as calling that function, you should instead do something like:

```ts
setInterval(() => clearAndReloadSpidStrategy(app, onRefresh).catch(_ => ...log the error...))
```

Reply:
fixed in a3bb176